Hw5

cidb123 opened the pull request:
Hw5.sql and Hw5.py are the two files ive been working on but i have more code in datagrip and pycharm not quite ready yet

cidb123 committed:
Uploading Homework

cidb123 committed:
Uploading Homework

cidb123 committed:
Uploading Homework

cidb123 committed:
Uploading Homework

cidb123 committed:
Uploading Homework

cidb123 committed:
Uploading Homework

cidb123 committed:
Uploading HW

cidb123 committed:
Uploading Homework, some not completed, still getting general idea

cidb123 committed:
Uploading Homework, alot not debugged, ran out of time. Runs with an error for me when intializing with a main fucntion, says I need java.

cidb123 committed:
Uploading Midterm. Couldn't understand HTML, but tables in pycharm are working.

cidb123 committed:
Delete test.csv

cidb123 committed:
Delete dataset_loader.py

cidb123 committed:
Uploading H5 (partially), still need pitching stats, and loading correct graphs and models, and finally turn everything into HTML.

sebastienbeauvais commented on Hw5.py:
```diff
@@ -0,0 +1,74 @@
+    """
+    )
+
+    df = pd.DataFrame(cur.fetchall())
```
I would suggest adding column names instead of numbers. Although it doesn't necessarily matter which predictor is the best, it would be more human readable.

sebastienbeauvais commented on Hw5.py:
```diff
@@ -0,0 +1,74 @@
+    cont_response = []
+
+    # important loop that determines categorical variables.
+    for i in df:
```
since we only have one dataset, why not set categorical columns to dtype category? This would allow for a super easy split

sebastienbeauvais commented on Hw5.py:
```diff
@@ -0,0 +1,74 @@
+    oo_df = pd.DataFrame(index=cont_col, columns=cont_col)
+
+    for c, d in cont_cont_pair:
+        corr_cont = df.loc[:, c].corr(df.loc[:, d], method="pearson")
```
Code is breaking here. Looks like your date column of '3' isnt read in properly and python is trying to do arithmetic. I would suggest to reform date to just be year.

sebastienbeauvais commented on Hw5.py:
```diff
@@ -0,0 +1,74 @@
+
+    oo_df = pd.DataFrame(index=cont_col, columns=cont_col)
+
+    for c, d in cont_cont_pair:
```
not sure why you are doing it this way. Is it so you only get unique pairs (hr/bat_avg and not bat_avg/hr)? If so I would look into masking. I was able to run your code by removing this loop and doing:
corr_cont = df.corr()
oo_df2 = corr_cont.stack....

sebastienbeauvais commented on Hw5.sql:
```diff
@@ -0,0 +1,40 @@
+use baseball;
```
SQL works well. Not sure why when you call the table in python all the column labels disappear. Otherwise, try getting some comparative stats between teams as Julien mentioned in class.

sebastienbeauvais reviewed (commented):
Overall a solid start. Python code has a few issues (check comments). SQL is solid but your models will benefit from comparative stats. look into to_html() function to get your tables into html pages for final output. Also, the fig2 was not outputting anything.

cidb123 committed:
Add files via upload

In order for this to run, you need to set the correct permissions when connecting to mariadb(user, password) and you need to have the css in the working directory, as well as have the correct path with the html. I couldn't figure out clickable links and putting many plots in a html file, so I don't have a lot of the plots necessary. I'm also not sure why I get an error with the main function, I think it has to do with M1.

cidb123 committed:
Delete Midterm.py

cidb123 committed:
Delete Hw5.sql

cidb123 committed:
Add files via upload

cidb123 committed:
Add files via upload

Final changes with more description, still need to figure out clickable links because otherwise it would take too much space and be messy.
